Bottom: b6c359845dddebe90ed497f1588d639c5664e6ed
Top:    2c2f089694b554b7d69f4721d868d400444df972
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-10 20:29:54 +0000

Add profiling to file load
















---

diff --git a/src/file.c b/src/file.c
index b6c67b9..018b272 100644
--- a/src/file.c
+++ b/src/file.c
@@ -350,6 +350,10 @@ LoadPCB (char *Filename)
 {
   PCBTypePtr newPCB = CreateNewPCB (False);
   Boolean units_mm;
+  clock_t start, end;
+  double elapsed;
+
+  start = clock();
 
   /* new data isn't added to the undo list */
   if (!ParsePCB (newPCB, Filename))
@@ -380,6 +384,10 @@ LoadPCB (char *Filename)
 	  CreateDefaultFont ();
 	}
 
+  end = clock();
+  elapsed = ((double) (end - start)) / CLOCKS_PER_SEC;
+  printf ("Loading file %s took %f\n", Filename, elapsed);
+
       /* clear 'changed flag' */
       SetChangedFlag (False);
       PCB->Filename = MyStrdup (Filename, "LoadPCB()");
